tool/tsh/common/tsh.go

```diff
@@ -1815,6 +1815,11 @@ func onLogin(cf *CLIConf) error {
 		}
 	}
 
+	// Provide a warning when the auth connector is SSO and the username is specified.
```

Comment:
Let's make this comment describe the _why_, since the _what_ is pretty easy to follow.
```suggestion
	// If the cluster is using single-sign on, providing the user name
	// with --user is likely a mistake, so display a warning.
```